New package: numen-0.7

johngebbie opened the pull request:
#### Testing the changes
- I tested the changes in this PR: **YES**

#### New package
- This new package conforms to the [package requirements](https://github.com/void-linux/void-packages/blob/master/CONTRIBUTING.md#package-requirements): **YES**

#### Local build testing
- I built this PR locally for my native architecture, (x86_64-glibc)

This is [my voice control](https://numenvoice.com) for handsfree computing, mainly for accessibility needs.

JohnGebbie commented:
this is ready to merge when vosk-api has been: https://github.com/void-linux/void-packages/pull/39015

JohnGebbie commented:
updated the template to version 0.3.
this is ready to merge when vosk-api has been: https://github.com/void-linux/void-packages/pull/39015

JohnGebbie commented:
just updated the branch

JohnGebbie commented:
I'm going to release a new version soon, so let's hold off

JohnGebbie commented:
just updated the branch. still ready for review

paper42 commented on srcpkgs/numen/files/numen/conf:
```diff
@@ -0,0 +1,7 @@
+# Example to run the service as a specific user:
```
this file has to be marked as a conf_file, otherwise it would be overwritten on the next update

JohnGebbie replied:
Thank you :)!

JohnGebbie commented:
That's me released version 0.5 and updated the template.
I deleted and uninstalled numen and most the dependencies and reinstalled numen and all seems well.
I also edited the service conf and reinstalled numen to check my modifications would be preserved and they were.

JohnGebbie commented:
All seems well and ready to merge.

JohnGebbie commented:
Hold off there seems to be a bug in a recent commit

paper42 commented on srcpkgs/numen/template:
```diff
@@ -0,0 +1,39 @@
+}
+
+post_install() {
+	mv "${DESTDIR}/usr/bin/numen" speech
```
what's the purpose of this line?

JohnGebbie replied:
numen is not just a go executable, it's a pipline with shell scripts too.
The go module name has to be `git.sr.ht/~geb/numen` but the go file is speech.go.
I have added `go_package="speech.go"` so it is compiled as `speech` which is a slight improvement I think.

(I also thought I could compile it to $wrksrc by with `do_install` instead of `post_install` but then it ends up in a directory in $GOPATH.)

JohnGebbie commented:
Just updated the branch. This is ready to merge if you think so.

Eloitor commented:
I tested it and `it works well for me! Thank you

JohnGebbie commented:
Thanks @Eloitor, that's great :) You'd be very welcome on the [matrix chat](https://matrix.to/#/#numen:matrix.org) if that's something use.

JohnGebbie commented:
just git pulled, still ready

JohnGebbie commented:
@paper42 There should be the next release in a month or two and I'll bump the vosk-api and dotool templates at the same time, but I'd still love have this merged as it is.

JohnGebbie commented:
This is ready but these should be merged first: https://github.com/void-linux/void-packages/pull/42091 https://github.com/void-linux/void-packages/pull/41536

JohnGebbie commented:
Tested again with the bumped dependencies and all good.

JohnGebbie commented:
what do you think @paper42?

JohnGebbie commented:
I'd still love to get this merged if possible.

meator commented:
@johngebbie Hey, why did you close this PR?

johngebbie commented:
Hi @meator, I closed this because I no longer use Void. I'd be very happy for someone to take this forward again though!
